### PR TITLE
A0-1609 Removed unwrap()'s in aleph-client

### PR DESCRIPTION
**Pull request by Marcin-Radecki**
# Description

Motivation: get rid of bare `unwrap()` in `aleph-client`. One can just transform them to `expect(msg)`, but we can do better. We can benefit from leveraging `subxt::Error`, ie passing `Result<T, subxt::Error> to the client of `aleph-client`. This can be simple done as `.map_err(|e: subxt::Error| e.into())`, where the return type is `anyhow::Result` that is used already in `aleph-client`. Not all public API of `aleph-client` was modified in order not to bloat this PR. 

Deliberately only minor version of `aleph-client` was bumped to not bump major version twice. This is fine since we don't publish `aleph-client` to crates.io.

## Type of change

Please delete options that are not relevant.

- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: aleph-client/src/connections.rs [hunk 1]
```diff
@@ -113,6 +114,18 @@ impl Connection {
 
         Ok(R::decode(&mut bytes.as_ref())?)
     }
+
+    pub async fn get_block_number_inner(
```
Comment (pmikolajczyk41) [inline aleph-client/src/connections.rs:118]: the name suggests that it should be a private method (?)

File: aleph-client/src/utility.rs [hunk 2]
```diff
@@ -8,64 +8,63 @@ use crate::{
...
-            .header(Some(block))
+            .finalized_head()
             .await
-            .unwrap()
-            .map(|h| h.number)
+            .map_err(|e| e.into())
+    }
+
+    async fn get_block_number(&self, block: BlockHash) -> anyhow::Result<Option<BlockNumber>> {
+        self.get_block_number_inner(Some(block)).await
```
Comment (pmikolajczyk41) [inline aleph-client/src/utility.rs:60]: if you convert `get_block_number_inner` into a private method, I would move it to this module

File: e2e-tests/src/test/finalization.rs [hunk 3]
```diff
@@ -10,12 +11,15 @@ pub async fn finalization() -> anyhow::Result<()> {
...
-    let finalized = connection.connection.get_finalized_block_hash().await;
+    let finalized = connection.connection.get_finalized_block_hash().await?;
     let finalized_number = connection
         .connection
         .get_block_number(finalized)
-        .await
-        .unwrap();
+        .await?
+        .ok_or(anyhow!(
+            "Failed to retrieve block number for hash {:?}",
```
Comment (pmikolajczyk41) [inline e2e-tests/src/test/finalization.rs:20]: nit: you can inline `finalized` into the literal, like `anyhow!("Failed to retrieve block number for hash {finalized:?}")`

**Review by pmikolajczyk41 — APPROVED** [review]
cool!

File: e2e-tests/src/test/finalization.rs (hunk 3 above)
Comment (Marcin-Radecki) [reply e2e-tests/src/test/finalization.rs]: done

File: aleph-client/src/utility.rs (hunk 2 above)
Comment (Marcin-Radecki) [reply aleph-client/src/utility.rs:74]: done

File: aleph-client/src/connections.rs (hunk 1 above)
Comment (Marcin-Radecki) [reply aleph-client/src/connections.rs]: done

**Pull request merged**